### PR TITLE
PROD-1439 #311 - Fix include param for platform document API

### PR DESCRIPTION
**Pull request by surajkrsingh**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes https://github.com/buddyboss/buddyboss-platform-api/issues/311.

### How to test the changes in this Pull Request:

1. Pull the changes
2.  Check endpoint - `site-url/wp-json/buddyboss/v1/document?include=10`  ( e.g- document id as 10 )
3. See the response

### Proof Screenshots or Video
https://www.awesomescreenshot.com/video/5920582?key=aca06b75b795e13ec5bb84f5f86e5423

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Fixed the include param for platform document API

### CODE REVIEW

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@surajkrsingh We have already **in** parameter so why need to add extra **include** here.

**Comment by surajkrsingh** [comment]
@chetansatasiya 
As per the API document, 
it should have the `include` param  - https://www.buddyboss.com/resources/api/#api-Document-GetBBDocuments

And also if we check the consistency with another API endpoint like groups, videos and media, has `include` param. 

I didn't find the `in` param working for the document endpoints and 
the customer has created an issue because they follow the documentation.

Please let me know, your thought on this.

**Review by chetansatasiya — DISMISSED** [review]

File: src/bp-document/bp-document-functions.php [hunk 1]
```diff
@@ -354,6 +354,7 @@ function bp_document_get( $args = '' ) {
 			'search_terms'        => false,        // Pass search terms as a string.
 			'privacy'             => false,        // privacy of document.
 			'exclude'             => false,        // Comma-separated list of activity IDs to exclude.
+			'include'             => false,        // Comma-separated list of activity IDs to include.
```
Comment (KartikSuthar) [inline src/bp-document/bp-document-functions.php:357]:
```suggestion
			'in'             => false,        // Comma-separated list of activity IDs to include.
```

File: src/bp-document/bp-document-functions.php [hunk 2]
```diff
@@ -379,6 +380,7 @@ function bp_document_get( $args = '' ) {
 			'scope'               => $r['scope'],
 			'privacy'             => $r['privacy'],
 			'exclude'             => $r['exclude'],
+			'include'             => $r['include'],
```
Comment (KartikSuthar) [inline src/bp-document/bp-document-functions.php:383]:
```suggestion
			'in'             => $r['in'],
```

File: src/bp-document/classes/class-bp-document.php [hunk 3]
```diff
@@ -811,6 +811,7 @@ public static function documents( $args = array() ) {
 				'sort'                => 'DESC',          // ASC or DESC.
 				'order_by'            => 'date_created',  // Column to order by.
 				'exclude'             => false,           // Array of ids to exclude.
+				'include'             => false,           // Array of ids to include.
```
Comment (KartikSuthar) [inline src/bp-document/classes/class-bp-document.php:814]: remove this

File: src/bp-document/classes/class-bp-document.php [hunk 4]
```diff
@@ -914,6 +915,13 @@ public static function documents( $args = array() ) {
 			$where_conditions_folder['exclude']   = "f.id NOT IN ({$exclude})";
 		}
 
+		// Include specified items.
+		if ( ! empty( $r['include'] ) ) {
+			$include                              = implode( ',', wp_parse_id_list( $r['include'] ) );
+			$where_conditions_document['include'] = "d.id IN ({$include})";
+			$where_conditions_folder['include']   = "f.id IN ({$include})";
+		}
+
```
Comment (KartikSuthar) [inline src/bp-document/classes/class-bp-document.php:924]: Remove this as already have `in`

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

**Comment by KartikSuthar** [comment]
@surajkrsingh 
Also need to update the API: https://github.com/buddyboss/buddyboss-platform-api/blob/release/includes/bp-document/classes/class-bp-rest-document-endpoint.php#L302
`$args['in'] = $request['include'];`

**Comment by surajkrsingh** [comment]
@KartikSuthar @chetansatasiya 
Changes updated on the PR

We already making changes here - https://github.com/buddyboss/buddyboss-platform/pull/3141#discussion_r744421185
Then it's not needed here - https://github.com/buddyboss/buddyboss-platform/pull/3141#issuecomment-962858716

Tested the API result demo
image

**Review by KartikSuthar — APPROVED** [review]

**Review by dipskakadiya — APPROVED** [review]

**Comment by dipskakadiya** [comment]
@surajkrsingh Approved. can you prepare this for QA.

**Comment by surajkrsingh** [comment]
Merged into `dev` and `qa-ongoing`

**Pull request closed without merging**